src/store/messages/saga.ts

```diff
@@ -477,6 +486,12 @@ export function* replaceOptimisticMessage(currentMessages, message) {
 export function* receiveUpdateMessage(action) {
   let { message } = action.payload;
 
+  // Check if the message already exists in the state
```

Comment:
Resolves a race condition where both user action and server event trigger a UI update, causing flickering/edited message renders as received message. Implemented a check against the Redux state to determine if a message has already been updated = preventing a redundant server event from triggering a UI refresh.